### ISSUE TITLE
Over Engineered and Validated Module - that does not work for tier points!

### ISSUE THREAD

**Issue by Ecoent** [MicrosoftDocs/dynamics-365-unified-operations-public#617]
can i just re-do everything?
I just want to try a Loyalty program for retail online first! (then offline).
Tried everything as per the documents.
UI is nice, but nothing works and everywhere there is "validation" that goes nowhere!

I just want to add loyalty points, lets say $0.10 per $100 purchase on tier1, then something similar for tier 2 (call it Gold), and tier 3 (Platinum). and see it gets reflected for basic 3 tier members (buyers).

Whereas this tool talks about a lot of discount codes and I get that its meant for a large retail chain company, but hey, online first!

Can someone pinpoint to a simple one pager document that does only loyalty points and how to see those in demo transactions?

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: d69dbedf-6ac0-c762-a732-f7c7e60bde07
* Version Independent ID: 1292ab05-4902-c8d1-b8e0-7cb9778d4415
* Content: [Loyalty overview - Retail | Dynamics 365](https://docs.microsoft.com/en-us/dynamics365/unified-operations/retail/set-up-customer-loyalty-program#loyalty-enhancements)
* Content Source: [articles/retail/set-up-customer-loyalty-program.md](https://github.com/MicrosoftDocs/Dynamics-365-Unified-Operations-Public/blob/live/articles/retail/set-up-customer-loyalty-program.md)
* Service: **dynamics-365-retail**
* GitHub Login: @scott-tucker
* Microsoft Alias: **scotttuc**

**Comment by ShalabhjainMSFT** [comment]
Hi Ecoent,

Thanks for trying out the loyalty capabilities. I am sorry to hear you did not get the desired results. I have tried to provide the steps below, hopefully these are helpful for you.

If you don't need special pricing and discounts for loyalty members, then ignore creating special price groups and discounts. Follow the below steps to set up a simple loyalty program that you can use for earning and redeeming points.

1. Set up loyalty payment method as one of the supported tender types for the retail store. This is not needed if you do not want to test redeeming the loyalty points
2. Set up date intervals. These are used to define tier rules, expiry date and vesting period for the reward points. For testing just create two date intervals e.g. NextYear, PrevYear which include 1 year in future and past one year duration respectively
3. Set up loyalty reward points that will be earned or redeemed. For e.g. set up a 'redeemable' loyalty reward point named "Test reward point" of type "Amount" and choose a currency.
4. Set up a loyalty program. Add three tiers e.g. Silver, gold and diamond. For each of these tiers choose the date interval as "NextYear". This means that once a customer qualifies for a tier, they will be qualified for one year.
5. For each tier set up a tier rule. e.g. for silver do not add a tier rule which means all new customers are silver. For Gold, set the rule as if the customer has earned aka 'Minimum issued' 100 "Test reward points", then they should be qualified for the Gold tier. for Diamond, set the rule as if the customer has earned aka 'Minimum issued' 500 "Test reward points", then they should be qualified for the Diamond tier. 
6. Set up a loyalty scheme for the loyalty program created above. 
-- Set up earning rules for different tiers e.g. on spending 1 USD the silver customer gets 1 "Test reward points". Similarly, set different earning rules for Gold and Diamond.
-- Set up redemption rules. I would recommend to have just one redemption rule e.g. 1 point = .25 USD. This means, that silver, gold and diamond tier customers earn differently, but the value of a point is same for all
-- Add one or more channels where this loyalty scheme should be applied e.g. choose Houston store if using demo data
7. Choose a number sequence for automatically generating loyalty card in D365FO and POS. To do so, provide a number sequence for "Card number". This is present under "Number sequences" tab of "Retail shared parameters"
8. The set up is complete, and now this data needs to be send to the channels. For that first run a job named "Process loyalty schemes" (Retail IT > Loyalty > Process loyalty schemes). Then run the job 9999 from the distribution schedule.

**Comment by Ecoent** [comment]
thanks for a quick response.
i am gonna try these steps out first thing next week, and post things here.

**Comment by Ecoent** [comment]
here it is: 
The configuration process seems like it could satisfy most of our current project needs, but will need
custom data processing, as our fundamental loyalty process is way different than typical market 
(legacy) type points. We do have points, but we just give more every month for each tier.


7) this is quite confusing for me, i did find a number sequence somewhere else:
    Retail > Channel setup > Payment Methods > Card Numbers 
    it shows "loyalty card" under Card ID column, and number from 100000 to 200000
    the other path that you suggested gives the actual field (looks like a database field id)
    So in Setup Retail Shared Parameters for number sequences:
   " Card_Number"  and field is "Reta_16"

Q 1 for (7):    You also mentioned D365FO and POS, do we need to also have financial operations and POS modules in our subscription?
What if we don't want POS module?

8) I could find and run "Process Loyalty Schemes" but could not find job 9999.
    the search box on top right is quite good (really good) to jump on different topics, but it did not
    find job 9999.

Q. 2) Overall, is there any developer account on Github that we can develop our own logic using SQL scripts or Javascripts and put calculations somewhere. Right now we run all our app programs on a hosted server that have MySQL and MS SQL, plus a custom developed app - for online ecommerce.
and there is another project getting designed on blockchain (IBM) will do similar with fiat currencies and cryptos in future.

We are looking to build a cloud solution (central hub kinda) that will serve for all our commerce shops (all online for now) including Shopify, few on Magento to use this as a repository for all customer loyalty.
Also, we want to connect our shoppers and third party (client portals, partners portal and more apps) shoppers on a secured login app, that will give them access to their rewards and points.
Some of our partner customers will ask for POS - which is not now but in future.

In all - we also need a developer account with more freedom to achieve this for online commerce first. 
Not sure whether we will go for a licensed account, if the costs are high. If you can pinpoint to few things about this - will be great help.
Please include a roadmap of development of Retail module as well.

Appreciate your help. 

Thanks
Ron

PS: added some screenshots.

![card number sequence ss2](https://user-images.githubusercontent.com/44745681/48774180-7ad81b00-ec97-11e8-9857-426308522cde.JPG)
![retail shared parameters ss1](https://user-images.githubusercontent.com/44745681/48774181-7b70b180-ec97-11e8-86ef-5322a3bba66d.JPG)

**Comment by ShalabhjainMSFT** [comment]
Apologies for the delay. 

1. For adding the number sequence you needed to navigate to the "Retail shared parameters" and choose a number sequence for "Card number" field. I see you have already done that by choosing "Reta_16".

2. Regarding the question on if POS is required - No, POS is not mandatory. You can earn and redeem points for online transactions as well

3. Regarding job 9999. You need to navigate to the "Distribution schedule" form and there you will find the job 9999. This job will send the information from the D365FO to all the channels i.e. online channel, B&M stores (POS) etc.

4. Just to make sure I understand your requirements, is it fair to say all you need is a mechanism to read the loyalty balance for a customer from the app that you are building? If yes, then it is possible by using the appropriate Retail server API's. I can find out which API you would have to use, once you confirm if that is what you need.

**Comment by Ecoent** [comment]
Thanks, honestly i thought this is forgotten by MS team. Also, i need to create another trial run, it ran out of 30 days.  Again, to recreate everything!!. 

to answer your question in (4) above: all we want to do is create a central place for points, that can connect to major CMS and ecommerce apps, and all customers can login to see their rewards
in this loyalty app, no matter what shop a customer is member of, they can login here.

(Online will also be followed by offline - seamless experience in future, as some retail chains
can benefit from ads, more loyal customers)

Major difference is we gonna give points every month to the consumer / customer on each purchase. Rather than one time by most loyalty program.

For shops our customers and we are using Magento, Shopify and Woocommerce / WP as CMS.
Also VueJS and ReactJS based front end. 

user actions are rewarded for shopping, referrals, social media and travel blogging activities. Our target market is English, Spanish and Chinese speaking customers, also will follow other languages in Europe, S East Asia.

to sum up,
1) For our existing shops, we need just the loyalty app in the beginning and not the entire microsoft 
    retail universe.
2) if any customer asks for help and depending on their willingness, we will offer them MS Dynamics.
    as a value add priority service. and shop front ends - that we can bundle are in Vue, React and
    popular CMS systems. We also offer backend and process automation bots.

3) rest, per the market data, i think more companies are willing to digitize things on their own 
   commerce, which is where MS Dynamics is a good suite of apps.

Hope this clarifies things.

Please suggest how we can test on Loyalty only and use it to develop real use cases without the trial period clock tick.

Thank
Ron
